packages/api-plugin-orders/src/mutations/placeOrder.js

```diff
@@ -112,173 +8,10 @@ async function createPayments({
-    paymentsInput,
-    shippingAddress: shippingAddressForPayments,
-    shop
-  });
-
-  // Create anonymousAccessToken if no account ID
-  const fullToken = accountId ? null : getAnonymousAccessToken();
-
-  const now = new Date();
-
-  const order = {
-    _id: orderId,
-    accountId,
-    billingAddress,
-    cartId,
-    createdAt: now,
-    currencyCode,
-    discounts,
-    email,
-    ordererPreferredLanguage: ordererPreferredLanguage || null,
-    payments,
-    shipping: finalFulfillmentGroups,
-    shopId,
-    surcharges: orderSurcharges,
-    totalItemQuantity: finalFulfillmentGroups.reduce((sum, group) => sum + group.totalItemQuantity, 0),
-    updatedAt: now,
-    workflow: {
-      status: "new",
-      workflow: ["new"]
-    }
-  };
-
-  if (fullToken) {
-    const dbToken = { ...fullToken };
-    // don't store the raw token in db, only the hash
-    delete dbToken.token;
-    order.anonymousAccessTokens = [dbToken];
-  }
-
-  let referenceId;
-  const createReferenceIdFunctions = getFunctionsOfType("createOrderReferenceId");
-  if (!createReferenceIdFunctions || createReferenceIdFunctions.length === 0) {
-    // if the cart has a reference Id, and no custom function is created use that
-    if (_.get(cart, "referenceId")) { // we want the else to fallthrough if no cart to keep the if/else logic simple
-      ({ referenceId } = cart);
-    } else {
-      referenceId = Random.id();
-    }
-  } else {
-    referenceId = await createReferenceIdFunctions[0](context, order, cart);
-    if (typeof referenceId !== "string") {
-      throw new ReactionError("invalid-parameter", "createOrderReferenceId function returned a non-string value");
-    }
-    if (createReferenceIdFunctions.length > 1) {
-      Logger.warn("More than one createOrderReferenceId function defined. Using first one defined");
-    }
-  }
-
-  order.referenceId = referenceId;
-
-
-  // Apply custom order data transformations from plugins
-  const transformCustomOrderFieldsFuncs = getFunctionsOfType("transformCustomOrderFields");
-  if (transformCustomOrderFieldsFuncs.length > 0) {
-    let customFields = { ...(customFieldsFromClient || {}) };
-    // We need to run each of these functions in a series, rather than in parallel, because
-    // each function expects to get the result of the previous. It is recommended to disable `no-await-in-loop`
-    // eslint rules when the output of one iteration might be used as input in another iteration, such as this case here.
-    // See https://eslint.org/docs/rules/no-await-in-loop#when-not-to-use-it
-    for (const transformCustomOrderFieldsFunc of transformCustomOrderFieldsFuncs) {
-      customFields = await transformCustomOrderFieldsFunc({ context, customFields, order }); // eslint-disable-line no-await-in-loop
-    }
-    order.customFields = customFields;
-  } else {
-    order.customFields = customFieldsFromClient;
-  }
+  const { appEvents, collections, userId } = context;
+  const { Orders } = collections;
```

Comment:
These two destructuring statements can be combined

Reply:
Done